module/VuFind/src/VuFind/Controller/AbstractSearch.php

```diff
@@ -842,17 +840,15 @@ protected function processAdvancedCheckboxes($params, $savedSearch = false)
         $section = $params[1] ?? 'CheckboxFacets';
 
         // Load config file:
-        $config = $this->getService(\VuFind\Config\PluginManager::class)
-            ->get($config);
+        $config = $this->getService(\VuFind\Config\ConfigManager::class)->getConfigArray($config);
 
         // Process checkbox settings in config:
         $flipCheckboxes = false;
         if (str_starts_with($section, '~')) {        // reverse flag
             $section = substr($section, 1);
             $flipCheckboxes = true;
         }
-        $checkboxFacets = ($section && isset($config->$section))
-            ? $config->$section->toArray() : [];
+        $checkboxFacets = ($section && isset($config['$section'])) ? $config['$section'] : [];
```

Comment:
I don't think you want these single quotes!

```suggestion
        $checkboxFacets = ($section && isset($config[$section])) ? $config[$section] : [];
```

...and if this was done by some kind of regex, you might want to double check that there aren't any other `'$` patterns that I missed elsewhere.

Reply:
Alarmingly, this does not appear to cause any tests to fail. I'll investigate and see if I can fill the coverage gap.

Reply:
See #4575 -- that new test passes on the dev branch but fails here.

Reply:
...and it's passing again now that you've fixed the issue.

Reply:
Good catch! That could have been hard to find later...